src/com/sap/piper/DefaultValueCache.groovy

```diff
@@ -29,4 +44,60 @@ class DefaultValueCache implements Serializable {
     static reset(){
         instance = null
     }
+
+
+    private static Set immutable(Set s) {
+        immutable(s, (Set)[])
+    }
+
+    private static List immutable(List l) {
+        immutable(l, (List)[])
+    }
+
+    private static def immutable(Collection _in, Collection _out) {
+
+        for(def e : _in) {
+            if(e in List || e in Set || e in Map) {
```

Comment:
Do we need to support `List` and `Set`? How is this defined in `yaml`?

Reply:
Set we actually do not need, I agree. It seems "out of the box" not possible to get sets when parsing json. --> Set support has been erased.